docs/app/pages/components/components-sections/input-controls/typeahead/snippets/app.ts

```diff
@@ -0,0 +1,55 @@
+import { Component } from '@angular/core';
+import { TypeaheadKeyService, TypeaheadOptionEvent } from '@ux-aspects/ux-aspects';
+import { BehaviorSubject } from 'rxjs';
+
+@Component({
+    selector: 'app',
+    templateUrl: 'app.component.html',
+    styleUrls: ['./app.component.css'],
+})
+export class AppComponent {
+
+    dropdownOpened: boolean = false;
+    selectOnEnter: boolean = false;
+    dropDirection: 'up' | 'down' = 'down';
+    selectFirst: boolean = false;
+
+    private _input$ = new BehaviorSubject<string>('');
+    private  _value$ = new BehaviorSubject<string>(null);
+
+    /** Load the options and filter the them */
+    loadOptions(pageNum: number, pageSize: number, filter: string) {
+
+        const displayValues = ['Alpha', 'Beta', 'Gamma', 'Delta', 'Epsilon', 'Zeta', 'Eta', 'Theta', 'Iota', 'Kappa', 'Lambda', 'Mu', 'Nu', 'Xi', 'Omicron', 'Pi', 'Rho', 'Sigma', 'Tau', 'Upsilon', 'Phi', 'Chi', 'Psi', 'Omega'];
```

Comment:
We should probably make these items a class value, eg:

![image](https://user-images.githubusercontent.com/20795331/62936529-f51a7380-bdc1-11e9-93ee-ed19da71a974.png)

You would need to add a reference to a function bound to the correct context, otherwise you'll get an error saying `this._values` is undefined.

It would probably good to make a longer list to demonstrate paging. Perhaps also add a small delay to simulate server loading time. Consider using `chance.js` to populate the list with lots of data for paging.

Reply:
Added <code>chance.js</code>